shared/src/store/action-reducers/simulated-region.ts

```diff
@@ -29,6 +31,95 @@ import { TransferPointActionReducers } from './transfer-point';
 import { isCompletelyLoaded } from './utils/completely-load-vehicle';
 import { getElement, getElementByPredicate } from './utils/get-element';
 
+/**
+ * This function assumes that every SimulatedRegion holds **ONLY** material,vehicles,patients and personnel
+ * (in any amount) and **ONE** TransferPoint.
+ * It only deletes material,vehicles and personnel if they are **ALL** in the SimulatedRegion to be deleted
+ * If one were to add more things to a SimulatedRegion one would have to change this function accordingly.
+ * @param draftState: The Draft State from which the SimulatedRegion should be deleted.
+ * @param simulatedRegionId: The Id of the SimulatedRegion that should be deleted.
+ */
+
+export function deleteSimulatedRegion(
+    draftState: Mutable<ExerciseState>,
+    simulatedRegionId: UUID
+) {
+    // Find related vehicles
+
+    const relatedVehicles = Object.values(draftState.vehicles).filter(
+        (vehicle) =>
+            isInSpecificSimulatedRegion(vehicle, simulatedRegionId) &&
+            Object.keys(vehicle.materialIds).every((materialId) =>
+                isInSpecificSimulatedRegion(
+                    getElement(draftState, 'material', materialId),
+                    simulatedRegionId
+                )
```

Comment:
It would also be okay if the material is in the vehicle (same for the personnel).

Reply:
Done